### PR TITLE
csrcolor

### PR DESCRIPTION
**Pull request by YvanMokwinski**
wrapper of csrcolor, won't compile until rocsparse-internal PR #215 is merged.

### CODE REVIEW

File: clients/tests/test_csrcolor.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * THE SOFTWARE.
+ *
+ * ************************************************************************ */
+
+#include "testing_csrcolor.hpp"
+
+#include <gtest/gtest.h>
+#include <hipsparse.h>
+
+TEST(csrcolor_bad_arg, csrcolor_bad_arg_float)
```
Comment (jsandham) [inline clients/tests/test_csrcolor.cpp:29]: Probably gate all these with:

// Only run tests for CUDA 11.1 or greater
#if(!defined(CUDART_VERSION) || CUDART_VERSION >= 11010)

Thats what we do for all the other tests. That way we limited the cuda versions the tests run for.

Comment (YvanMokwinski) [reply clients/tests/test_csrcolor.cpp:31]: fixed

**Comment by YvanMokwinski** [comment]
won't compile 'til rocSPARSE-internal PR #223 being merged.

File: clients/include/testing_csrcolor.hpp [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+    hipsparseStatus_t status;
+
+    std::unique_ptr<handle_struct> unique_ptr_handle(new handle_struct);
+    hipsparseHandle_t              handle = unique_ptr_handle->handle;
+
+    std::unique_ptr<descr_struct> unique_ptr_descr(new descr_struct);
+    hipsparseMatDescr_t           descr = unique_ptr_descr->descr;
+
+    auto m_coloring    = hipsparse_unique_ptr{device_malloc(sizeof(int) * 1), device_free};
```
Comment (jsandham) [inline clients/include/testing_csrcolor.hpp:62]: Why multiply by 1?

Comment (YvanMokwinski) [reply clients/include/testing_csrcolor.hpp:62]: because we don't care, this bad arg.

File: clients/common/hipsparse_template_specialization.cpp [hunk 3]
```diff
@@ -8310,4 +8310,116 @@ namespace hipsparse
...
+                                  csrRowPtrA,
+                                  csrColIndA,
+                                  fractionToColor,
+                                  ncolors,
+                                  coloring,
+                                  reordering,
+                                  info);
+    }
+
 } // namespace hipsparse
```
Comment (ntrost57) [inline clients/common/hipsparse_template_specialization.cpp:8425]: Guess we can also copy the rocsparse way here at some point

**Review by ntrost57 — APPROVED** [review]

**Review by jsandham — APPROVED** [review]

**Pull request merged**